icefoganalytics/travel-authorization
# Travel Authorization Editing

Fixes https://github.com/icefoganalytics/travel-authorization/issues/3

# User Story

As a user, I want to edit an existing travel authorization request, _when request is as a draft status_ so that I can make necessary changes before final submission.

I should be able to edit these fields:
- travel steps
- travel method
- final destination
- purpose of travel

## TODO

- [x] support saving original submission as draft
- [x] lock edits to only draft submissions
- [x] test edit pipeline around "travel details section"

# Context

You can view existing travel requests at http://localhost:8080/my-travel-requests.
This renders the `src/web/src/modules/travelForm/views/FormList.vue` component.
The edit view is `src/web/src/modules/travelForm/views/TravelFormEdit.vue`, which is separate from the create view at `src/web/src/modules/travelForm/views/TravelFormCreate.vue`

# QA Instructions

1. Boot the database service via `dev up`
2. Boot the api service via `cd src/api && npm run start`
3. Boot the web service via `cd src/web && npm run start`
4. Go to http://localhost:8080/my-travel-requests.
5. Click on the "+ Travel Authorization" button.
6. Complete the "Enter your personal details" section.
7. Complete the "Tell us about the travel" section.
8. Complete the "Enter details about the trip and purpose" section.
9. Click the "Save Draft" button. This will redirect you back to the http://localhost:8080/my-travel-requests page.
10. Click on the newly created travel request (bottom of the list), and you will be redirected to the "edit" page for that request at http://localhost:8080/my-travel-requests/:travel-request-id
11. Click through the form, which should look the same as the create form, and make sure that the answers you provided are correctly filled in in edit mode.
12. Make some edits and check that they save. Each field should be checked, in theory, but for practical purposes at least check one field for each section.

Check security:
1. Create a new travel authorization request but use the "Submit to Supervisor" button.
2. Go back to the newly submitted form, and check that all edits are rejected by the back-end (403).
> A future ticket should involve building out a separate "read only" component for forms that are not editable, and directing the user there instead.
